### PR TITLE
[Calling] Send reaction component calling-component-binding and composite changes

### PR DESCRIPTION
**Pull request by Mohtasim**
# What

The change contains the following package updates:
* calling-component-bindings
* react-composites
* communication-react

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 1]
```diff
@@ -855,6 +856,13 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
     });
   }
 
+  /* @conditional-compile-remove(reaction) */
+  public async onReactionClicked(emoji: string): Promise<void> {
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:860]: we should call the param here: `reaction` especially since if Contoso is to provide their own sprites like talked about in option one we should make sure that we are making them map to that as well.

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 2]
```diff
@@ -401,6 +403,10 @@ export const CallControls = (props: CallControlsProps & ContainerRectProps): JSX
                 <RaiseHand displayType={options?.displayType} />
               )
           }
+          {
+            /* @conditional-compile-remove(reaction) */
+            <Reaction displayType={options?.displayType} />
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:408]: We should make sure that this button can be hidden through the CallControl options like the other buttons in the control bar.

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatBackedCallAdapter.ts [hunk 3]
```diff
@@ -48,6 +48,9 @@ export class CallWithChatBackedCallAdapter implements CallAdapter {
   constructor(callWithChatAdapter: CallWithChatAdapter) {
     this.callWithChatAdapter = callWithChatAdapter;
   }
+  onReactionClicked(emoji: string): Promise<void> {
+    throw new Error(`Reaction of type ${emoji} send not successful.`);
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatBackedCallAdapter.ts:52]: Is there a reason that we are throwing here? believably we need to make sure that we are able to use reactions in the CallWithChatComposite?

File: packages/calling-component-bindings/src/callControlSelectors.ts [hunk 4]
```diff
@@ -168,6 +170,36 @@ export const raiseHandButtonSelector: RaiseHandButtonSelector = reselect.createS
...
+ * Selector for {@link ReactionButton} component.
+ *
+ * @beta
+ */
+export const reactionButtonSelector: ReactionButtonSelector = reselect.createSelector(
+  [getLocalParticipantReactionState, getCallState],
+  (reaction, callState) => {
+    return {
+      checked: reaction ? true : false,
+      disabled: callState === 'InLobby' ? true : callState === 'Connecting' ?? false
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/calling-component-bindings/src/callControlSelectors.ts:198]: in fact is it only enabled when call state is connected?

File: packages/calling-component-bindings/src/videoGallerySelector.ts [hunk 5]
```diff
@@ -61,7 +61,7 @@ export type VideoGallerySelector = (
 export const videoGallerySelector: VideoGallerySelector = createSelector(
   [
     getScreenShareRemoteParticipant,
-    getRemoteParticipantsExcludingConsumers,
+    getRemoteParticipants,
```
Comment (JamesBurnside) [inline packages/calling-component-bindings/src/videoGallerySelector.ts:64]: Why was this change needed, I believe we intentionally are filtering out consumers

File: packages/react-composites/src/composites/CallComposite/components/buttons/Reaction.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  styles?: ControlBarButtonStyles;
+  disabled?: boolean;
+}): JSX.Element => {
+  const reactionButtonProps = usePropsFor(ReactionButton) as unknown as ReactionButtonProps;
+  const callStatus = useSelector(getCallStatus);
+  const styles = useMemo(() => concatButtonBaseStyles(props.styles ?? {}), [props.styles]);
+
+  let reactionButtonDisabled = isDisabled(props.option);
+
+  if (_isInLobbyOrConnecting(callStatus)) {
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/components/buttons/Reaction.tsx:36]: Is this needed? Looks like the selector is handling this

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatTypeAssertions.ts [hunk 7]
```diff
@@ -27,7 +27,10 @@ import { FileUploadAdapter } from '../../ChatComposite';
 
 /// CallWithChatAdapterManagement
 
-type CallWithChatAdapterManagementInternal = Omit<CallAdapterCallManagement, 'removeParticipant'> &
+type CallWithChatAdapterManagementInternal = Omit<
+  CallAdapterCallManagement,
+  'removeParticipant' | 'onReactionClicked'
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatTypeAssertions.ts:32]: this shouldn't be needed here, CallWithChatComposite adapter should handle emojis the same as calladapter

**Review by JamesBurnside — COMMENTED** [review]
Add the label `update-snapshots` to generate the new composite UI snapshots that will have the reaction button in the control bar

File: packages/calling-component-bindings/src/callControlSelectors.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Mohtasim) [reply packages/calling-component-bindings/src/callControlSelectors.ts:198]: Yes. It should be enabled when call state in connected.
Updating accordingly.

File: packages/calling-component-bindings/src/videoGallerySelector.ts (hunk 5 above)
Comment (Mohtasim) [reply packages/calling-component-bindings/src/videoGallerySelector.ts:64]: I see. It got refactored with merge conflicts. Updating it. Thanks :)

File: packages/react-composites/src/composites/CallComposite/components/buttons/Reaction.tsx (hunk 6 above)
Comment (Mohtasim) [reply packages/react-composites/src/composites/CallComposite/components/buttons/Reaction.tsx:36]: Well, the selector was not handling the lobby one. But with the refactoring, can only disable through selector. Hence removing this method. Thanks :)

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png:1]: Looks like mobile shouldn't have the chevron

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png:1]: Should it be in the more options on mobile?

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png:1]: Actually looking at figma, should it ever have the chevron?
image

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png:1]: I see mobile it should be completely different, feel free to hide on mobile for this PR until the seprate UI is created
image

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/call-with-chat-custom-buttons-Desktop-Chrome-linux.png
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/call-with-chat-custom-buttons-Desktop-Chrome-linux.png:1]: Can we hide reactions button on this test

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png:1]: Can be a separate PR, but looks like the reactions icon needs to have the same fill/non-fill beahvior as raise hand:
image

So looks like two things:
- Reactions should be before raise hand
- Raise hand icon should be updated to be the non-fill version

File: packages/calling-component-bindings/src/callControlSelectors.ts [hunk 8]
```diff
@@ -168,6 +170,36 @@ export const raiseHandButtonSelector: RaiseHandButtonSelector = reselect.createS
...
+ * Selector for {@link ReactionButton} component.
+ *
+ * @beta
+ */
+export const reactionButtonSelector: ReactionButtonSelector = reselect.createSelector(
+  [getLocalParticipantReactionState, getCallState],
+  (reaction, callState) => {
+    return {
+      checked: reaction ? true : false,
+      disabled: callState === 'Connected' ? true : false
```
Comment (JamesBurnside) [inline packages/calling-component-bindings/src/callControlSelectors.ts:198]: reverse this, when it's connected it should be false (i.e. not disabled)

```suggestion
      disabled: callState !== 'Connected'
```

File: packages/communication-react/review/beta/communication-react.api.md [hunk 9]
```diff
@@ -1848,6 +1854,9 @@ export type CommonCallControlOptions = {
     raiseHandButton?: boolean | {
         disabled: boolean;
     };
+    reactionButton?: boolean | {
+        disables: boolean;
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:1858]: sp: disables -> disabled

File: packages/calling-component-bindings/src/callControlSelectors.ts (hunk 8 above)
Comment (Mohtasim) [reply packages/calling-component-bindings/src/callControlSelectors.ts:198]: I dont think the selector implementation is taking effect. Because this is clearly wrong logic and and it still worked.

Comment (Mohtasim) [reply packages/calling-component-bindings/src/callControlSelectors.ts:198]: Thanks. Updating this.

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png
Comment (Mohtasim) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png:1]: This gives a compile time error saying - An object literal cannot have multiple properties with the same name in strict mode.
Open for any suggestion on this.

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png:1]: Dang, you'll probably have to define a new component for this then, just at the bottom of the file, can you try something like this:
```tsx
const ControlButtonRaiseHandIcon = (): JSX.Element => {
  /* @conditional-compile-remove(reactions) */
  return <HandRight20Regular />,
  return <HandRight20Filled />;
}
```

then:
```tsx
ControlButtonRaiseHand: <ControlButtonRaiseHandIcon />,
```

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/call-with-chat-custom-buttons-Desktop-Chrome-linux.png
Comment (Mohtasim) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/call-with-chat-custom-buttons-Desktop-Chrome-linux.png:1]: reaction and raise hand has been hidden on this test.

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png
Comment (Mohtasim) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/CallControls.test.ts-snapshots/user-set-control-bar-button-options-Desktop-Chrome-linux.png:1]: Updated.

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png
Comment (Mohtasim) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/callwithchat/hermetic/Call.test.ts-snapshots/call-with-chat-gallery-screen-Mobile-Android-Landscape-linux.png:1]: Mobile view has been hidden for now. Mobile view will be made with a different PR.

File: packages/react-composites/src/composites/common/ControlBar/CommonCallControlBar.tsx [hunk 10]
```diff
@@ -316,6 +318,16 @@ export const CommonCallControlBar = (props: CommonCallControlBarProps & Containe
                         disableTooltip={props.mobileView}
                       />
                     )}
+                    {
+                      /* @conditional-compile-remove(reaction) */
+                      !props.mobileView && isEnabled(options.raiseHandButton) && (
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/common/ControlBar/CommonCallControlBar.tsx:323]: Can you explain this section? I see we add the reaction button below the RaiseHand also, shouldonly one of these exist?
Also I assume this isEnabled line should be isEnabled(options.reactionsButton) and not isEnabled(options.raiseHandButton)?
```suggestion
                      !props.mobileView && isEnabled(options.reactionsButton) && (
```

Comment (JamesBurnside) [reply packages/react-composites/src/composites/common/ControlBar/CommonCallControlBar.tsx:323]: Ah yeah looking at the snapshots, we should delete that second `<Reaction` usage below as there are two reaction buttons:
image

Comment (Mohtasim) [reply packages/react-composites/src/composites/common/ControlBar/CommonCallControlBar.tsx]: Ops. This might have added due to merge conflicts. Removed the extra section. Thanks :)

File: packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/CallControls.test.ts-snapshots/no-mics-Desktop-Chrome-linux.png
[the remark this answers is not in the document]
Comment (Mohtasim) [reply packages/react-composites/tests/browser/snapshots/beta/tests/browser/call/hermetic/CallControls.test.ts-snapshots/no-mics-Desktop-Chrome-linux.png:1]: Removing the chevron altogether now according to figma.

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatTypeAssertions.ts (hunk 7 above)
Comment (Mohtasim) [reply packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatTypeAssertions.ts:32]: Pushing the change here in the subsequent PR with capability PR.

**Review by dmceachernmsft — APPROVED** [review]
